qutebrowser/browser/downloads.py

```diff
@@ -695,12 +718,11 @@ def fetch(self, reply, fileobj=None, filename=None, auto_remove=False):
         """
         if fileobj is not None and filename is not None:
             raise TypeError("Only one of fileobj/filename may be given!")
-        if filename is not None:
-            suggested_filename = os.path.basename(filename)
-        elif fileobj is not None and getattr(fileobj, 'name', None):
-            suggested_filename = fileobj.name
-        else:
-            _inline, suggested_filename = http.parse_content_disposition(reply)
+        if not suggested_filename:
+            if filename is not None:
+                suggested_filename = os.path.basename(filename)
+            elif fileobj is not None and getattr(fileobj, 'name', None):
+                suggested_filename = fileobj.name
```

Comment:
Turns out there actually *is* a minor issue with passing `suggested_filename` here instead of calling `http.parse_content_disposition`: If `fetch()` gets called with `suggested_filename`/`filename`/`fileobj` not set, it'll display an empty prompt even though it *could* get that information from the reply.

This is currently only triggered when retrying a failed download I think. That one would be easy to fix (and I need to rewrite how download retries are done because of #502), but I wonder if the drawback is really worth it. After all, other places could also use `fetch()` in the future, and then all the callers would have to use `http.parse_content_disposition` by themselves.

If I'm not mistaken this is only a cosmetic change, right? Then it might make sense to revert this.

Sorry for not catching this earlier!